service/taker.go

```diff
@@ -110,79 +110,82 @@ func (s *Service) SwapStarted(ctx context.Context, swapId uuid.UUID, txHash stri
 	for _, frag := range swap.Frags {
 		// get order by ID
-		order, err := s.orderBookStore.FindOrderById(ctx, frag.OrderId, false)
+		order, err := store.FindOrderById(ctx, frag.OrderId, false)
 		// no return during erros as what can be revert, should
 		if err != nil {
 			logctx.Error(ctx, "order not found while reverting a swap", logger.Error(err))
 		} else if !validatePendingFrag(frag, order) {
 			logctx.Error(ctx, "Swap fragments should be valid during a revert request", logger.Error(err))
 		} else {
-			// success
+			// success - Ulock an order even if cancelled so the condition IsPending below would be False and the order will be removed
 			logctx.Debug(ctx, "Unlock Fragment", logger.String("orderID", frag.OrderId.String()), logger.String("OutSize", frag.OutSize.String()))
 			err = order.Unlock(ctx, frag)
 			if err != nil {
 				logctx.Error(ctx, "Unlock Failed", logger.Error(err))
 				return err
 			}
-			s.publishOrderEvent(ctx, order)
-			orders = append(orders, *order)
+			// no need to publish nor update cancelled order
+			if !order.Cancelled {
+				if svc != nil {
+					svc.publishOrderEvent(ctx, order)
+				}
+				// save to modify/update new pending state in db
+				unlockedOrders = append(unlockedOrders, *order)
+			}
 		}
-		// cancelled orders
-		if order != nil && !order.Cancelled {
-			ordersToCancel = append(ordersToCancel, *order)
+		// remove cancelled unfilled non pending unlocked orders
+		if order != nil && order.Cancelled && !order.IsPartialFilled() && !order.IsPending() {
+			ordersToRemove = append(ordersToRemove, *order)
 		}
 	}
-	// store orders
-	err = s.orderBookStore.StoreOpenOrders(ctx, orders)
+	err := store.PerformTx(ctx, func(txid uint) error {
+		// update unlocked orders
+		for _, order := range unlockedOrders {
+			if err := store.TxModifyOrder(ctx, txid, models.Update, order); err != nil {
+				logctx.Error(ctx, "AbortSwap Failed updating unlocked order", logger.Error(err), logger.String("orderId", order.Id.String()))
+				return err
+			}
+		}
+		// remove order from all entries
+		for _, order := range ordersToRemove {
+			if err := store.TxRemoveOrder(ctx, txid, order); err != nil {
+				logctx.Error(ctx, "TxRemoveOrder Failed", logger.Error(err), logger.String("id", order.Id.String()))
+				return err
+			}
+		}
+		return nil
+	})
+	return err
+
+}
+
+func (s *Service) AbortSwap(ctx context.Context, swapId uuid.UUID) error {
+	logctx.Debug(ctx, "AbortSwap", logger.String("swapId", swapId.String()))
+	// get swap from store
+	swap, err := s.orderBookStore.GetSwap(ctx, swapId, true)
+
 	if err != nil {
-		logctx.Warn(ctx, "StoreOrders Failed", logger.Error(err))
+		logctx.Warn(ctx, "GetSwap Failed", logger.Error(err))
 		return err
 	}
-	// remove cancelled orders
-	err = s.cancelUnlockedOrders(ctx, ordersToCancel)
+
+	// mutual impl for ABORT and RESOLVE(false) swap
+	err = unlockSwapAndHandleCancelledOrders(ctx, s, s.orderBookStore, swap)
```

Comment:
General suggestion: abstract away implementation details like you have done here more often. It makes it so much easier to reason about (rather than reading lines and lines of implementation details)